### PR TITLE
Fix `SeqPathChunk` 

### PR DESCRIPTION
**Pull request by lihaoyi**
Fixes https://github.com/com-lihaoyi/os-lib/issues/159

### CODE REVIEW

File: os/src/Path.scala [hunk 1]
```diff
@@ -43,9 +43,9 @@ object PathChunk {
   implicit class SeqPathChunk[T](a: Seq[T])(implicit f: T => PathChunk) extends PathChunk {
     var segments0 = Nil
     var ups0 = 0
```
Comment (lefou) [inline os/src/Path.scala:45]: Any idea, why these are here? Could you add some `@deprecated` on them, so we can remove them in the next major bump?

File: os/test/src/PathTests.scala [hunk 2]
```diff
@@ -425,5 +425,10 @@ object PathTests extends TestSuite{
         }
       }
     }
+    test("issue159"){
+      val result = os.rel / Seq(os.up, os.rel/"hello", os.rel/"world")
+      val expected = os.up / "hello" / "world"
+      assert(result == expected)
```
Comment (lefou) [inline os/test/src/PathTests.scala:431]: Would be nice to also cover the `ArrayPathChunk` case here, as it's also affected by #159.

**Comment by lihaoyi** [comment]
@lefou do you know what to do with these `os.jvm[3.1.3].checkFormat Found 1 misformatted files` errors? Is there some command I can run to autoformat things compliantly?

**Comment by lefou** [comment]
> @lefou do you know what to do with these `os.jvm[3.1.3].checkFormat Found 1 misformatted files` errors? Is there some command I can run to autoformat things compliantly?

Yeah. You can run:

```
$ mill __.reformat
```

**Pull request merged**